### PR TITLE
Render Markdown in option docs for flakes and longDescription

### PR DESCRIPTION
**Pull request by ncfavier**
Closes https://github.com/NixOS/nixos-search/issues/535, closes https://github.com/NixOS/nixos-search/issues/525

To do:

- [x] We're currently creating a new temporary file containing the xref lua filter for *each* Pandoc invocation. This is bad. I think it would generally be nicer to package the files we need (pandoc filters, `flake_info.nix`) alongside `flake-info`: we could set a `DATADIR` variable in the Nix build that is detected at compile time and, if unset, defaults to `CARGO_MANIFEST_DIR` so that `cargo run` still finds the files.
- [x] The `pandoc` crate does not let us define precisely the flavour of Markdown we want ([Commonmark with a few extensions](https://nixos.org/manual/nixpkgs/stable/#sec-contributing-markup)) because it does not expose all of Pandoc's formats and extensions (notably `commonmark_x` and `+fenced_divs`). For now I've set the format to `markdown`, which is hopefully a superset, but if finer control is needed we will have to ~invoke pandoc directly~ or fork the crate. At least manpage references seem to render fine. Ping @jtojnar @pennae
- [x] There seems to be a regression with `nix-security`. EDIT: this is a bug exposed by making `FlakeEntry` tagged: we don't support multiple licenses in flake packages, so previously flake packages with a list of licenses were parsed as *apps* instead.

### CODE REVIEW

File: flake-info/src/data/pandoc.rs [hunk 1]
```diff
@@ -63,8 +52,34 @@ impl<T: AsRef<str>> PandocExt for T {
...
+            PandocOutput::ToBuffer(html) => {
+                format!("<rendered-html>{}</rendered-html>", html)
+            }
+            _ => unreachable!(),
+        })
+    }
+
+    fn render_markdown(&self) -> Result<String, PandocError> {
+        let roles_filter = Path::new(FILTERS_PATH).join("myst-reader/roles.lua");
+        let man_filter = Path::new(FILTERS_PATH).join("link-unix-man-references.lua");
```
[the remark this answers is not in the document]
Comment (ncfavier) [reply flake-info/src/data/pandoc.rs:64]: But we're writing HTML, not DocBook.

[the remark this answers is not in the document]
Comment (ncfavier) [reply flake-info/src/data/pandoc.rs:64]: Right, we didn't do anything like that.

**Comment by ncfavier** [comment]
Ah, `markdown` requires a blank line before lists whereas `commonmark` doesn't. That settles it, let's switch to commonmark.

**Comment by ncfavier** [comment]
I've added Markdown rendering for `longDescription`.

As a reminder/note to self, XSS sanitisation is done by Elm.

**Comment by ncfavier** [comment]
@ysndr ping

**Comment by ncfavier** [comment]
(or @garbas)

**Review by ysndr — APPROVED** [review]

**Comment by ysndr** [comment]
Looks good, and brings in some welcome cleanup
Thank you @ncfavier
Is there a way to see the result in the preview?

**Comment by ncfavier** [comment]
> Is there a way to see the result in the preview?

You can look at [tmux](https://632899fb7bd5b614cc23836f--nixos-search.netlify.app/packages?channel=unstable&show=tmux&from=0&size=50&sort=relevance&type=packages&query=tmux) or [geomyidae](https://632899fb7bd5b614cc23836f--nixos-search.netlify.app/flakes?channel=unstable&show=services.geomyidae.package&from=0&size=50&sort=relevance&type=options&query=geo).

**Comment by ncfavier** [comment]
Ah, I shouldn't have merged so soon. The nixpkgs-unstable import is currently broken (but a [fix](https://github.com/NixOS/nixpkgs/pull/198247) is underway), so that channel will be rolled back a bit until the next update.

Also it seems like the flake import is hitting GitHub API rate limits.

**Pull request merged**